### ISSUE TITLE
Jetson Xavier NX usable?

### ISSUE THREAD

**Issue by MerlinBrasil** [ethereum-mining/ethminer#2158]
I've been searching high and low for an effective miner to use for the Jetson Xavier NX.

Is ethminer optimized for this card? Either Linux or Windows would be fine.

With such a low power draw (15 V) I'll be using 4 or 8 cards to complete the rig.

Thanks,
Merlin

**Comment by jean-m-cyr** [comment]
Some time ago I managed to get Ethminer to build and run on the Jetson nano. A word of advice: Xavier will make a poor mining platform. It's CUDA cores have no local memory and rely on the hosts memory accessing it via PCIe bridge. Very slow.

**Comment by MerlinBrasil** [comment]
Thanks, Jean
Disappointing news, but very good to know. ;)
Best,
Merlin

**Comment by MerlinBrasil** [comment]
Jean, I had one more question for you. Obviously the Nano and the Xavier NX are different. Why do you believe the results would be equally dismal?

**Comment by jean-m-cyr** [comment]
They both share the same memory architecture, namely no high speed memory directly connected to the CUDA cores. This is not a problem if computations can be kept to internal CUDA core registers. Ethash requires heavy access to a little over 4GB (and growing) of RAM and it all has to happen over PCIe on Jetson or Xavier.

**Comment by jmtrappier** [comment]
They state memory bandwidth is 51.2 Go/s (8 Go 128-bit LPDDR4x @ 51,2 Go/s), is this not fast enough ?

**Comment by jean-m-cyr** [comment]
That is bandwidth to the CPU, not the GPU.

**Comment by jmtrappier** [comment]
Hey Jean (are you french ?),

I may test on a Jetson AGX Xavier, could you provide me some information on how to use it on this hardware so I can test ?

**Comment by jean-m-cyr** [comment]
Last I tried, I just followed the standard cmake procedure under Ubuntu on the Jetson. I built for ARM just fine out of the box. Was limited in what it could mine since the Jetson has only 4GB of ram. The ETH DAG size alone is more than 4GB.

**Comment by jmtrappier** [comment]
Hello,

The AGX have more memory, reason why I would like to test it.

Will keep you updated.

On 19/02/2021 18:08:39, Jean Cyr <notifications@github.com> wrote:
Last I tried, I just followed the standard cmake procedure under Ubuntu on the Jetson. I built for ARM just fine out of the box. Was limited in what it could mine since the Jetson has only 4GB of ram. The ETH DAG size alone is more than 4GB.

**Comment by SimonTheVillain** [comment]
> 
> 
> Some time ago I managed to get Ethminer to build and run on the Jetson nano. A word of advice: Xavier will make a poor mining platform. It's CUDA cores have no local memory and rely on the host's memory accessing it via PCIe bridge. Very slow.

Nope. Actually both CPU and GPU, both access RAM via some "Memory Controller Fabric" they both can achieve the same memory bandwidth. [see](https://developer.nvidia.com/blog/jetson-xavier-nx-the-worlds-smallest-ai-supercomputer/)

**Comment by jean-m-cyr** [comment]
Well the Nano only achieved about 50KH/S. It'll be interesting to see how the Xavier does.

**Comment by jean-m-cyr** [comment]
The Dagger Hashimoto hash used by ETH is intentionally memory constrained such that the relationship between hash rate and memory bandwidth is virtually linear. So, a little math,

A tuned Vega 64 GPU achieves 50MH/S at 125watts, and has a memory bandwidth of 483.8 GB/s. The Xavier memory is rated at a theoretical max. of 137GB/s, so it should deliver up to 14MH/S at 15 watt.

I'd be very surprised....

**Comment by chamboin** [comment]
Hi, I'm trying to mine on Xavier AGX, but I conquer a building problem
```
CMake Error at CMakeLists.txt:71 (find_package):
  Could not find a package configuration file provided by "ethash" with any
  of the following names:

    ethashConfig.cmake
    ethash-config.cmake

  Add the installation prefix of "ethash" to CMAKE_PREFIX_PATH or set
  "ethash_DIR" to a directory containing one of the above files.  If "ethash"
  provides a separate development package or SDK, be sure it has been
  installed.
```
Can someone help?

**Comment by bobby20180331** [comment]
> The Dagger Hashimoto hash used by ETH is intentionally memory constrained such that the relationship between hash rate and memory bandwidth is virtually linear. So, a little math,
> 
> A tuned Vega 64 GPU achieves 50MH/S at 125watts, and has a memory bandwidth of 483.8 GB/s. The Xavier memory is rated at a theoretical max. of 51GB/s, so it should deliver up to 5MH/S at 15 watts.
> 
> I'd be very surprised....
Hi
@jean-m-cyr 

I have copiled on nvidia jetson Xavier AGX. which has a memory bandwidth of about 137GB/s. but hash rate have only about 0.2MH/s. what's wrong with it?

**Comment by bobby20180331** [comment]
> The Dagger Hashimoto hash used by ETH is intentionally memory constrained such that the relationship between hash rate and memory bandwidth is virtually linear. So, a little math,
> 
> A tuned Vega 64 GPU achieves 50MH/S at 125watts, and has a memory bandwidth of 483.8 GB/s. The Xavier memory is rated at a theoretical max. of 51GB/s, so it should deliver up to 5MH/S at 15 watts.
> 
> I'd be very surprised....
Hi
@jean-m-cyr 

I have copiled on nvidia jetson Xavier AGX. which has a memory bandwidth of about 137GB/s. but hash rate have only about 0.2MH/s. what's wrong with it? 
https://www.nvidia.cn/autonomous-machines/embedded-systems/

**Comment by chamboin** [comment]
> > The Dagger Hashimoto hash used by ETH is intentionally memory constrained such that the relationship between hash rate and memory bandwidth is virtually linear. So, a little math,
> > A tuned Vega 64 GPU achieves 50MH/S at 125watts, and has a memory bandwidth of 483.8 GB/s. The Xavier memory is rated at a theoretical max. of 51GB/s, so it should deliver up to 5MH/S at 15 watts.
> > I'd be very surprised....
> > Hi
> > @jean-m-cyr
> 
> I have copiled on nvidia jetson Xavier AGX. which has a memory bandwidth of about 137GB/s. but hash rate have only about 0.2MH/s. what's wrong with it? My interested point is how to achieve 10~20MH/s on xavier AGX platform.
> 
> ![image](https://user-images.githubusercontent.com/37145604/110603989-8e8a5100-81c2-11eb-9d4a-f0954cdbdba3.png)
> 
> https://www.nvidia.cn/autonomous-machines/embedded-systems/

Hi @bobby20180331 , I have compiled on my AGX Xavier platform as well, same hash rate with you, only 200kH/s, notify me if you find any interesting improvments please :)

**Comment by Bvgo** [comment]
Hi, 

How can I mine on Jetson nano, I need a step by step direction or a tutorial I could follow. Also how can I connect to multiple nodes running in different locations. 

It doesn't necessarily have to be ethereum. Thanks
